### PR TITLE
LIVE-2516: Related content [iOS and Android]

### PR DESCRIPTION
**Pull request by faresite**
## Why are you doing this?

The related content carousel is problematic for Android, so for the time being we're limiting the amount of cards to 2. 

## Changes

- add platform css classes clientside (`.js-android` and `.js-ios`)
- only display 2 cards for mobile Android and keep 4 cards for tablet screens.

## Screenshots

| Android|
| --- |
|  |

| Responsive |
| --- |
|  |

### CODE REVIEW

**Comment by benwuersching** [comment]
@faresite Thank you for the screenshots. In line with the Figma design, please can we make a few adjustments. See my notes below, also see the attached image (which is also on the Figma file).

1. Please reduce the top padding above ‘Related stories’ header to 2px. 

2. The header should read ‘Related stories’, with lowercase ‘s’ in stories. 

3. Please give the card top border a width of 1px (i.e. 3 times thicker than now). Also, please check the border colour which seems wrong. Border thickness and colour should follow the usual pillar rules and look exactly like on front cards. 

4. Please reduce the top padding above the card headline to 2px. Reduce the left/right padding to 8px. Again, the card should look exactly like a front card. 

5. Please reduce the padding between headline and date stamp to 16px.  

6. Reduce the right padding of the date stamp to 8px. 

7. Please reduce the padding below the date stamp to 4px.  

8. Increase the font-weight of the date stamp to 700. 

9. Please check the image ratio. Our images have a ration of 5:3 but the ones in the screenshot seem to have a somewhat different ratio. 

10. There seems to be a thin white border below the image that should not be there. 

11. Please reduce the padding between the card bottom and the copyright line to 16px.
 
12. For a temporary responsive solution for how many cards we’re showing as the device width increases, let’s do this:  
- smallest: 2 cards, 8px padding between cards 
- from 660px width: 3 cards, 8px padding
 - from 740px width: 4 cards, 16px padding
 
 Many thanks!
 
adjustments

**Comment by faresite** [comment]
@benwuersching I've managed to work through the majority of these now, a couple of problems came up:

1. **_Please reduce the top padding above ‘Related stories’ header to 2px. _**
done ✅

2. **_The header should read ‘Related stories’, with lowercase ‘s’ in stories. _**
done ✅

3. **_Please give the card top border a width of 1px (i.e. 3 times thicker than now). Also, please check the border colour which seems wrong. Border thickness and colour should follow the usual pillar rules and look exactly like on front cards. _**
The card top border is currently 1px, so I wonder if it's the screenshot fault that you think it's less than that. The border colour is following the related item pillar, so unless we have a bug somewhere this should not be wrong. Do you think any of these cards in the screenshot should have a different border colour? Might it be you think the colours are wrong because the camera icon colour is different than the top border colour? If that's the case, the reason they're different is because the icon uses the "inverted" colour from the palette and the top border uses the "kicker" colour, not sure why it's this way but maybe that's the problem that I need to fix?

4. **_Please reduce the top padding above the card headline to 2px. Reduce the left/right padding to 8px. Again, the card should look exactly like a front card. _**
done ✅

5. **_Please reduce the padding between headline and date stamp to 16px.  _**
Currently the related cards are given a min height of 10rem. The problem with giving a padding here is that cards will have different heights based on the length of the headline. The way I see this happening in fronts is that there's a set minimum padding for the longest card headline and the other cards would adjust to that. This might be a bit difficult to do in CSS hence it's the reason it was given a min height of 10rem. I want to try and see what can be done to make it better but wanted to clarify this first.

6. **_Reduce the right padding of the date stamp to 8px. _**
done ✅

7. **_Please reduce the padding below the date stamp to 4px._**
done ✅

8. **_Increase the font-weight of the date stamp to 700. _**
done ✅

9. **_ Please check the image ratio. Our images have a ratio of 5:3 but the ones in the screenshot seem to have a somewhat different ratio. _**
WIP and will depend a bit on issue no.5

10. **_There seems to be a thin white border below the image that should not be there. _**
WIP and will depend a bit on issue no.5

11. **_Please reduce the padding between the card bottom and the copyright component to 16px._**
done ✅

12. **_For a temporary responsive solution for how many cards we’re showing as the device width increases, let’s do this:  
• smallest: 2 cards, 8px padding between cards 
• from 480px width: 3 cards, 8px padding
• from 740px width: 4 cards, 16px padding_**
WIP

**Comment by benwuersching** [comment]
@faresite Thank you for your comments, and thank you for working through all the issues. 
Regarding (3.) the card top border colour, I think you're right. It was just the screenshot that made the thickness and colour look wrong. I just checked the GIF you uploaded, and on the GIF it all looks good. So disregard the whole thing :). 
Regarding (5.) the overall card height and spacing between header and date stamp: On what do we base 1 rem? Is it body text size (i.e. 17px), or is it 16px?

**Comment by JamieB-gu** [comment]
> On what do we base 1 rem? Is it body text size (i.e. 17px), or is it 16px?

16px

**Comment by faresite** [comment]
@benwuersching I was experimenting a bit and I found a way of making all the cards t he same height with a minimum padding of 16px for the longest headline, similar behaviour to apps cards. Not sure how the camera icon should be though as I'm used to it being more symmetrical (left and bottom padding equal). Here it has a 8px left padding and 4px bottom padding.
Screenshot 2021-06-22 at 10 40 12
 Here's the full screenshot:

Screenshot 2021-06-22 at 10 38 35

**Comment by benwuersching** [comment]
@faresite The new screenshot looks great. Thank you. About the alignment of the camera/media icon: best to put it 8px from left and 8px from the bottom.

**Comment by benwuersching** [comment]
@faresite I wanted to tell you about a change of mind I had about the font-weight we should use for the headlines on cards (changing from Semibold to Medium, in line with Source). Funnily enough, I checked your screenshots and you seem to have built it with Medium **(font-weight 500)** anyway. So if indeed that's the case, then that's great: nothing to do/change on your end. 😃

**Comment by faresite** [comment]
## Android screenshots

Screenshot 2021-06-24 at 10 20 34
Screenshot 2021-06-24 at 10 20 51
Screenshot 2021-06-24 at 10 21 30
Screenshot 2021-06-24 at 10 22 05
Screenshot 2021-06-24 at 10 22 32

**Comment by faresite** [comment]
## iOS 

@benwuersching @prisalcalde 

Screenshot 2021-06-24 at 11 04 44
Screenshot 2021-06-24 at 11 04 52
Screenshot 2021-06-24 at 11 05 17
Screenshot 2021-06-24 at 11 05 23
Screenshot 2021-06-24 at 11 05 37

**Comment by faresite** [comment]
## Android

You can see the font size fixed here
Screenshot 2021-06-24 at 11 12 51
Screenshot 2021-06-24 at 11 12 55
Screenshot 2021-06-24 at 11 13 03
Screenshot 2021-06-24 at 11 13 12

File: config/cloudformation.yaml [hunk 1]
```diff
@@ -176,7 +176,7 @@ Resources:
         - Key: App
           PropagateAtLaunch: true
           Value: !Ref App
-      VPCZoneIdentifier: !Split [ ',', !ImportValue MobileAppsApiVPC-Subnets]
+      VPCZoneIdentifier: !Split [ ',', !ImportValue MobileAppsApiVPC-PublicSubnets]
```
Comment (JamieB-gu) [inline config/cloudformation.yaml:179]: Why are these changes included? Has there been a rebase/merge against #1351 that's got muddled somewhere?

File: src/client/setup.ts [hunk 2]
```diff
@@ -27,10 +27,22 @@ function performanceMetrics(): void {
 	);
 }
 
+function platformCSS(): void {
+	const getPlatformClass = (): string => {
+		const ua = navigator.userAgent;
+		if (/android/i.test(ua)) {
```
Comment (JamieB-gu) [inline src/client/setup.ts:33]: I suspect this might be brittle, but it's a temporary workaround for a bug so 👍 for now.

File: src/components/shared/card.tsx [hunk 3]
```diff
@@ -53,13 +54,13 @@ const listStyles = (
...
-		margin-right: ${remSpace[3]};
-		flex: 0 0 15rem;
-		height: 100%;
+		margin-right: ${remSpace[2]};
+		flex: 0 0 42vw;
 		display: flex;
 		flex-direction: column;
 		justify-content: space-between;
 		border-top: ${borderColor(type, format)};
+		max-width: 160px;
```
Comment (JamieB-gu) [inline src/components/shared/card.tsx:63]: `10rem`?

File: src/components/shared/card.tsx [hunk 4]
```diff
@@ -68,6 +69,18 @@ const listStyles = (
 		${darkModeCss`
             background: ${neutral[7]};
         `}
+
+		${from.tablet} {
+			margin-right: ${remSpace[5]};
+		}
+
+		${from.desktop} {
+			max-width: 220px;
```
Comment (JamieB-gu) [inline src/components/shared/card.tsx:78]: `13.75rem`?

File: src/components/shared/card.tsx [hunk 5]
```diff
@@ -119,29 +136,35 @@ const anchorStyles = css`
...
     `}
+	display: flex;
+	flex-direction: column;
+	height: 100%;
 `;
 
 const headingWrapperStyles = css`
-	padding: ${remSpace[3]};
-	min-height: 10rem;
+	padding: 2px ${remSpace[2]} ${remSpace[4]};
```
Comment (JamieB-gu) [inline src/components/shared/card.tsx:145]: This padding doesn't fit onto the design system spacing chart @benwuersching? If not can we make it `0.125rem` @faresite?

File: src/components/shared/card.tsx [hunk 6]
```diff
@@ -268,6 +292,7 @@ const iconStyles = (format: Format): SerializedStyles => {
 		display: inline-block;
 		background-color: ${inverted};
 		border-radius: 50%;
+		margin-top: -4px;
```
Comment (JamieB-gu) [inline src/components/shared/card.tsx:295]: Possibly `- remSpace[1]`? I realise SVG alignment is tricky so 👍 if this doesn't work.

File: src/components/shared/relatedContent.tsx [hunk 7]
```diff
@@ -35,6 +36,49 @@ const listStyles = css`
...
+			li {
+				flex: 1 1 33.33%;
+			}
+			li:nth-child(3) {
+				display: flex;
+			}
+		}
+		${from.tablet} {
+			li {
+				flex: 0 0 160px;
```
Comment (JamieB-gu) [inline src/components/shared/relatedContent.tsx:67]: `10rem`?

File: src/components/shared/relatedContent.tsx [hunk 8]
```diff
@@ -35,6 +36,49 @@ const listStyles = css`
...
+			}
+			li:nth-child(n + 3) {
+				display: flex;
+			}
+			li:nth-child(n + 2) {
+				margin-left: 20px;
```
Comment (JamieB-gu) [inline src/components/shared/relatedContent.tsx:73]: `$remSpace[5]`?

File: src/components/shared/relatedContent.tsx [hunk 9]
```diff
@@ -35,6 +36,49 @@ const listStyles = css`
...
+			}
+		}
+		${from.desktop} {
+			li {
+				flex: 0 0 220px;
```
Comment (JamieB-gu) [inline src/components/shared/relatedContent.tsx:78]: `13.75rem`?

File: src/styles.ts [hunk 10]
```diff
@@ -88,13 +88,13 @@ export const lineStyles = css`
...
 	background: ${neutral[97]};
-	margin: ${remSpace[3]};
+	margin: 0 ${remSpace[3]};
 
 	${darkModeCss`
         background: ${background.inverse};
     `};
 
-	padding: ${remSpace[6]} 0;
+	padding: 2px 0 0;
```
Comment (JamieB-gu) [inline src/styles.ts:97]: Same question about the design system and usage of `rem`.

**Review by JamieB-gu — COMMENTED** [review]
Nice! Some comments, mostly about use of `px` over `rem`. You may run into scaling issues with `px`.

File: config/cloudformation.yaml (hunk 1 above)
Comment (faresite) [reply config/cloudformation.yaml:179]: needed to merge this in yesterday so I can deploy to CODE, I won't include those before merging this PR 👍

File: src/components/shared/card.tsx (hunk 3 above)
Comment (faresite) [reply src/components/shared/card.tsx:63]: Would that not a problem with the font resizer? The cards could end up being massive?

File: src/components/shared/card.tsx (hunk 5 above)
Comment (faresite) [reply src/components/shared/card.tsx:145]: This padding is per @benwuersching comments above. I can remify it though 👍

File: src/components/shared/relatedContent.tsx (hunk 7 above)
Comment (faresite) [reply src/components/shared/relatedContent.tsx:67]: same question as above

**Review by JamieB-gu — APPROVED** [review]

**Comment by benwuersching** [comment]
@faresite The screenshots look good, thank you!

**Comment by prisalcalde** [comment]
Thanks for adding the screenshots here, @faresite . Looks great to me!
